lib/gui/components/update-notifier/services/update-notifier.js

```diff
@@ -16,10 +16,24 @@
 
 'use strict';
 
+const _ = require('lodash');
 const semver = require('semver');
 const etcherLatestVersion = require('etcher-latest-version');
 
-module.exports = function($http, $q, ModalService, UPDATE_NOTIFIER_SLEEP_TIME, ManifestBindService, SettingsModel) {
+module.exports = function($http, $q, ErrorService, ModalService, UPDATE_NOTIFIER_SLEEP_TIME, ManifestBindService, SettingsModel) {
+
+  // Memoize the version Promise
+  const memoizeVersion = (func) => {
```

Comment:
Looks like you can just provide a "hash function" like `_.constant('latest-version')` as the second argument and `_.memoize` works fine with Promises. I've done that now.